src/test/scala/scala/collection/immutable/TestLazyListExtensions.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.immutable
+
+import org.junit.Assert._
+import org.junit.Test
+
+import next._
+
+class TestLazyListExtensions {
+
+  // This method will *not* terminate for non-cyclic infinite-sized collections.
+  def assertConstantMemory[T](xs: LazyList[T]): Unit =
+    // `force` does cycle detection, so if this terminates, the collection is
+    // either finite or a cycle
+    xs.force
+
+  @Test
+  def testEmpty1(): Unit = {
+    val xs = LazyList.empty  // realized
+    val cyc = xs.cycle
+    assertTrue(cyc.isEmpty)
+    assertTrue(cyc.size == 0)
+    assertEquals(Nil, cyc.toList)
+  }
+  @Test
+  def testEmpty2(): Unit = {
+    val xs = LazyList()  // not realized
```

Comment:
I have bad news for you... (see [LazyList.scala#L1156](https://github.com/scala/scala/blob/v2.13.4/src/library/scala/collection/immutable/LazyList.scala#L1156))

```scala
scala> LazyList().knownSize
val res1: Int = 0
```

you might try `LazyList(1).drop(1)`?

Reply:
I'll go with `LazyList.empty #::: LazyList.empty` and add a `knownSize` check